src/commands/deploy.ts

```diff
@@ -319,12 +322,31 @@ export default class Deploy extends Command {
     }
   }
 
+  async poll(deployment_id: string, match_status: string) {
+    return new Promise((resolve, reject) => {
+      let poll_count = 0;
+      const poll = setInterval(async () => {
+        const { data: deployment } = await this.architect.get(`/deploy/${deployment_id}`);
+        if (deployment.status.includes('failed') || poll_count > 100) {
+          clearInterval(poll);
+          reject(new Error('Deployment failed'));
+        }
+        if (deployment.status === match_status) {
+          clearInterval(poll);
+          resolve(deployment);
+        }
+        poll_count += 1;
+      }, 3000);
+    });
+  }
+
```

Comment:
Currently have most of this deploy logic copied in both cmds. I'll eventually consolidate into a helper class. Mostly just wanted to do the multi-line commend.